Sources/Common/DataModel/DataSetAttributes/FieldData.js

```diff
@@ -90,36 +98,106 @@ function vtkFieldData(publicAPI, model) {
         let destArr = publicAPI.getArrayByName(arr.getName());
         if (!destArr) {
           if (fromId < 0 || fromId > arr.getNumberOfTuples()) {
+            // TBD: should this be a deep or a shallow copy?
             publicAPI.addArray(arr);
+            other.getAttributes(arr).forEach((attrType) => {
+              publicAPI.setAttribute(arr, attrType);
+            });
           } else {
             const ncomps = arr.getNumberOfComponents();
             let newSize = arr.getNumberOfValues();
             const tId = toId > -1 ? toId : fromId;
-            if (newSize < tId * ncomps) {
+            if (newSize <= tId * ncomps) {
               newSize = (tId + 1) * ncomps;
             }
             destArr = vtkDataArray.newInstance({
               name: arr.getName(),
               dataType: arr.getDataType(),
-              numberOfComponents: arr.getNumberOfComponents(),
-              size: newSize,
+              numberOfComponents: ncomps,
+              values: macro.newTypedArray(arr.getDataType(), newSize),
+              size: 0,
             });
-            destArr.setTuple(tId, arr.getTuple(fromId));
+            destArr.insertTuple(tId, arr.getTuple(fromId));
             publicAPI.addArray(destArr);
+            other.getAttributes(arr).forEach((attrType) => {
+              publicAPI.setAttribute(destArr, attrType);
+            });
           }
         } else if (
           arr.getNumberOfComponents() === destArr.getNumberOfComponents()
         ) {
           if (fromId > -1 && fromId < arr.getNumberOfTuples()) {
             const tId = toId > -1 ? toId : fromId;
-            destArr.setTuple(tId, arr.getTuple(fromId));
+            destArr.insertTuple(tId, arr.getTuple(fromId));
+          } else {
+            // if `fromId` is not provided, just copy all (or as much possible)
+            // from `arr` to `destArr`.
+            destArr.insertTuples(arr.getTuples());
+          }
+        } else {
+          vtkErrorMacro('Unhandled case in passData');
+        }
+      }
+    });
+  };
+
+  publicAPI.interpolateData = (
+    other,
+    fromId1 = -1,
+    fromId2 = -1,
+    toId = -1,
```

Comment:
move up `toId` just after `other` to be consistent with C++ and limit errors

Reply:
Where did you find `interpolateData` in the C++ code ? I can't find it.

Reply:
it's `vtkDataSetAttributes::InterpolateEdge`, no ?

Reply:
That has a signature like this: `void vtkDataSetAttributes::InterpolateEdge(vtkDataSetAttributes* fromPd, vtkIdType toId, vtkIdType p1, vtkIdType p2, double t)`

(https://github.com/Kitware/VTK/blob/master/Common/DataModel/vtkDataSetAttributes.cxx#L1107)

There is no `vtkDataSetAttributes::InterpolateData`, however it is referenced in a comment here: https://github.com/Kitware/VTK/blob/master/Common/DataModel/vtkArrayListTemplate.txx#L172

(I assume that comment means `vtkDataSetAttributes::InterpolateAllocate`?)

I think a from - to order makes most sense, no?

Reply:
vtkDataSetAttributes methods have different from-to orderings between methods.

Let's leave it as is.